core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -41,6 +49,7 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
 
   private wordCloudElement: d3.Selection<SVGGElement, unknown, HTMLElement, any> | undefined;
   private c3ChartElement: c3.ChartAPI | undefined;
+  private map: mapboxgl.Map | undefined;
```

Comment:
As I have mentioned in another PR (https://github.com/Texera/texera/pull/1096#discussion_r623548455) we should modulize different charts (word cloud, line chart, and now scatterplot), instead of overwhelming everything to `VisualizationPanelContentComponent`.

Reply:
this can be handled in a future PR when I remove everything out of this class as we won't have a `Visualize` button and a popup

Reply:
Sounds good. Ideally please leave TODOs and comments in code.

Reply:
Agree with what Yicong said, please add a TODO comment in the code